### ISSUE TITLE
Negative gradient confusion

### ISSUE THREAD

**Issue by shashacks** [DigiRL-agent/digiq#12]
In the Digi-Q paper, you reference Tajwar et al. (2024) and say the negative gradient is detrimental to learning process. However, Tajwar et al. (2024) mention in Section 5.2 and Figures 12 and 14 that negative gradients ca be helpful for learning. I’m a bit confused by this discrepancy and would appreciate some clarification. When experimenting with Digi-Q, did you observe any performance decrease when using negative gradients?

**Comment by BiEchi** [comment]
Yes, negative gradients does not generally work effectively on LLM/VLM, and we observed it in this case. This is because LLMs/VLMs are trained with the next-token prediction task, which is essentially a 0/1 loss (learn or don't learn). Negative gradient introduces (1) a negative and (2) non-binary loss, like -0.4*logprobs, which conflicts its pretraining phase, thus causing difficulty during optimization.

**Comment by BiEchi** [comment]
Closing due to inactivity.
